executor/src/lib.rs

```diff
@@ -1,67 +1,60 @@
 extern crate console_error_panic_hook;
 
-use futures::prelude::*;
-use jsonrpsee::core::client::Client;
+use smoldot::json_rpc::methods::{HashHexString, HexString};
 use std::collections::BTreeMap;
 use wasm_bindgen::prelude::*;
 
+mod bindings;
 mod proof;
-mod runner_api;
 mod task;
-use crate::runner_api::RpcApiClient;
-use smoldot::json_rpc::methods::{HashHexString, HexString};
 
-#[wasm_bindgen]
-pub async fn get_metadata(code: &str) -> Result<JsValue, JsValue> {
+fn setup_console() {
     std::panic::set_hook(Box::new(console_error_panic_hook::hook));
-    _ = console_log::init_with_level(log::Level::Debug);
-
-    let code = HexString(hex::decode(&code[2..]).map_err(|e| e.to_string())?);
-    let metadata = task::Task::get_metadata(code)
-        .await
-        .map_err(|e| e.to_string())?;
-
-    Ok(metadata.to_string().into())
+    #[cfg(feature = "logging")]
+    {
+        let _ = console_log::init_with_level(log::Level::Trace);
+    }
 }
 
 #[wasm_bindgen]
-pub async fn get_runtime_version(code: &str) -> Result<JsValue, JsValue> {
-    std::panic::set_hook(Box::new(console_error_panic_hook::hook));
-    _ = console_log::init_with_level(log::Level::Debug);
+pub async fn get_runtime_version(code: JsValue) -> Result<JsValue, JsError> {
```

Comment:
could we just pass a callback object for query storage instead of using global variables? otherwise it may be hard to support running multiple networks in a same nodejs process

Reply:
If we're going to run multiple nodes in the same process then one solution will be to have an id for the chain and pass that around 
> could we just pass a callback object for query storage instead of using global variables? otherwise it may be hard to support running multiple networks in a same nodejs process

Reply:
that will also work. it is I always want to avoid global variable if possible

Reply:
I think callback is possible as well. Will see which is best

Reply:
will address into different PR